Tests/Pcap++Test/Tests/FileTests.cpp

```diff
@@ -751,3 +751,40 @@ PTF_TEST_CASE(TestPcapFileReadLinkTypeIPv4)
 
 } // TestPcapFileReadLinkTypeIPv4
 
+PTF_TEST_CASE(TestSolarisSnoopFileRead)
+{
+	pcpp::SnoopFileReaderDevice readerDev(EXAMPLE_SOLARIS_SNOOP);
+	PTF_ASSERT_TRUE(readerDev.open());
+	pcpp::RawPacket rawPacket;
+	int packetCount = 0;
+	int ethCount = 0;
+	int ipCount = 0;
+	int tcpCount = 0;
+	int udpCount = 0;
+	while (readerDev.getNextPacket(rawPacket))
+	{
+		packetCount++;
+		pcpp::Packet packet(&rawPacket);
+		if (packet.isPacketOfType(pcpp::Ethernet))
+			ethCount++;
+		if (packet.isPacketOfType(pcpp::IPv4))
+			ipCount++;
+		if (packet.isPacketOfType(pcpp::TCP))
+			tcpCount++;
+		if (packet.isPacketOfType(pcpp::UDP))
+			udpCount++;
+	}
+
+	pcpp::IPcapDevice::PcapStats readerStatistics;
+
+	readerDev.getStatistics(readerStatistics);
+	PTF_ASSERT_EQUAL((uint32_t)readerStatistics.packetsRecv, 250);
+	PTF_ASSERT_EQUAL((uint32_t)readerStatistics.packetsDrop, 0);
+
+	PTF_ASSERT_EQUAL(ethCount, 142);
```

Comment:
This file contains 250 packets, I wonder why Eth count isn't 250? 😳

Reply:
The remaining packets are EthernetDot3.  I extended the snoop testcase in commit 1e32b15 with check for 108 EthernetDot3 frames.